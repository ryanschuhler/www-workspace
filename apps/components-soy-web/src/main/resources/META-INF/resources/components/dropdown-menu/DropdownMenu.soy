{namespace DropdownMenu}

/**
 * DropdownMenu component
 */
{template .render}
  {@param buttonInnerHtml: html}
  {@param items: list<html|string>}
  {@param? cssClass: string}
  {@param? buttonCssClass: string}

  {let $buttonAttributes kind="attributes"}
    class="mdc-button
      {sp}mdc-menu-toggle

      {if $buttonCssClass}
        {sp}{$buttonCssClass}
      {/if}
    "

    onClick="handleOpenMenu_"
    type="button"
    ref="mdcMenuToggle"
  {/let}

  <div class="mdc-menu-anchor {$cssClass ?: ''}">
    <button {$buttonAttributes}>{$buttonInnerHtml}</button>

    <div class="mdc-simple-menu" data-mdc-auto-init="MDCSimpleMenu" ref="mdcSimpleMenu">
      <ul class="mdc-simple-menu__items mdc-list" role="menu" aria-hidden="true">
        {foreach $htmlItem in $items}
          <li class="mdc-list-item" role="menuitem" tabindex="0">
            {$htmlItem}
          </li>
        {/foreach}
      </ul>
    </div>
  </div>
{/template}