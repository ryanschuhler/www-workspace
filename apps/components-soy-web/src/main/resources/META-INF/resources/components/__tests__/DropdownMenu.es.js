'use strict';

import DropdownMenu from '../dropdown-menu/DropdownMenu.es';

let component;

const buttonInnerHtml = 'test-button-content';

const items = [
  '<p>item1</p>',
  '<p>item2</p>',
  '<p>item3</p>'
];

describe('DropdownMenu', () => {
  afterEach(() => {
    if (component) {
      component.dispose();
    }
  });

  beforeAll(() => {
		window.Liferay = {
			mdc: {
				menu: {
					MDCSimpleMenu: jest.fn()
				}
			},
			once: jest.fn()
    };
  });

  it('should render with three items', () => {
    component = new DropdownMenu({
      buttonInnerHtml: buttonInnerHtml,
      items: items
    });

    expect(component).toMatchSnapshot();

    expect(component.items.length).toBe(3);
  });
});