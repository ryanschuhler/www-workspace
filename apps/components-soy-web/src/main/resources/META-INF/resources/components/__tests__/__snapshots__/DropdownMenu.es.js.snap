// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenu should render with three items 1`] = `
<div class="mdc-menu-anchor ">
  <button class="mdc-button mdc-menu-toggle" type="button" ref="mdcMenuToggle">test-button-content</button>
  <div class="mdc-simple-menu" data-mdc-auto-init="MDCSimpleMenu" ref="mdcSimpleMenu">
    <ul class="mdc-simple-menu__items mdc-list" role="menu" aria-hidden="true">
      <li class="mdc-list-item" role="menuitem" tabindex="0">&lt;p&gt;item1&lt;/p&gt;</li>
      <li class="mdc-list-item" role="menuitem" tabindex="0">&lt;p&gt;item2&lt;/p&gt;</li>
      <li class="mdc-list-item" role="menuitem" tabindex="0">&lt;p&gt;item3&lt;/p&gt;</li>
    </ul>
  </div>
</div>
`;
